modules/nf-core/genrich/main.nf

```diff
@@ -8,34 +8,36 @@ process GENRICH {
         'biocontainers/genrich:0.6.1--h5bf99c6_1' }"
 
     input:
-    tuple val(meta), path(treatment_bam)
-    path  control_bam
+    tuple val(meta), path(treatment_bam), path(control_bam)
     path  blacklist_bed
     val   save_pvalues
     val   save_pileup
     val   save_bed
     val   save_duplicates
 
     output:
-    tuple val(meta), path("*narrowPeak")                     , emit: peaks
+    tuple val(meta), path("*.narrowPeak")                    , emit: peak
     tuple val(meta), path("*pvalues.bedGraph"), optional:true, emit: bedgraph_pvalues
-    tuple val(meta), path("*pileup.bedGraph") , optional:true, emit: bedgraph_pileup
-    tuple val(meta), path("*intervals.bed")   , optional:true, emit: bed_intervals
-    tuple val(meta), path("*duplicates.txt")  , optional:true, emit: duplicates
+    tuple val(meta), path("*pileup.bedGraph"),  optional:true, emit: bedgraph_pileup
+    tuple val(meta), path("*intervals.bed"),    optional:true, emit: bed_intervals
+    tuple val(meta), path("*duplicates.txt"),   optional:true, emit: duplicates
     path "versions.yml"                                      , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: ''
-    def prefix = task.ext.prefix ?: "${meta.id}"
-    def control    = control_bam    ? "-c $control_bam"               : ''
-    def blacklist  = blacklist_bed  ? "-E $blacklist_bed"             : ""
-    def pvalues    = save_pvalues   ? "-f ${prefix}.pvalues.bedGraph" : ""
-    def pileup     = save_pileup    ? "-k ${prefix}.pileup.bedGraph"  : ""
-    def bed        = save_bed       ? "-b ${prefix}.intervals.bed"    : ""
+    def args       = task.ext.args   ?: ""
+    def prefix     = task.ext.prefix ?: "${meta.id}"
+    def layout     = (!args.contains("-y") && meta.single_end) ? "-y" : ""
```

Comment:
What is this doing?

Reply:
Yes, would like also to know

Reply:
Single-end data can only be analyzed if unpaired alignments are kept (`-y` or `-w <int>`) so it defaults to `-y` if neither is already in `args` ([https://github.com/jsh58/Genrich/issues/77#issuecomment-876073734](https://github.com/jsh58/Genrich/issues/77#issuecomment-876073734)).